presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -191,6 +191,16 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
      */
     BlockEncoding getEncoding();
 
+    /**
+     * Create a new block from the current block by keeping the same elements
+     * only with respect to {@code visiblePositions}.
+     * May return a view over the data in this block or may return a copy
+     */
+    default Block getPositions(int[] positions)
```

Comment:
Why not use `int[] positions, int offset, int length` signature so that the method is future proof?

Reply:
That's a fair point. Somehow it would be nice to have `checkValidPositionsArray` in #9392. I skimmed through the patch and didn't find any major conflicts; so we could (1) land this after #9392, or (2) I will merge this patch first then create another PR to address this on top of #9392.

Reply:
Both are fine with me.